exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -337,28 +337,27 @@ func addSingleIntHistogramDataPoint(pt *otlp.IntHistogramDataPoint, metric *otlp
 	countlabels := createLabelSet(pt.GetLabels(), nameStr, baseName+countStr)
 	addSample(tsMap, count, countlabels, metric)
 
-	// count for +Inf bound
-	var totalCount uint64
+	// cumulative count for conversion to cumulative histogram
+	var cumulativeCount uint64
 
 	// process each bound, ignore extra bucket values
 	for index, bound := range pt.GetExplicitBounds() {
 		if index >= len(pt.GetBucketCounts()) {
 			break
 		}
```

Comment:
Check in advance that the size of the bucket counts is correct relative to the number of bounds. Return an error if not ok.